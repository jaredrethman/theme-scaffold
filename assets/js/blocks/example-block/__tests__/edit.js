/**
 * Example Component - Tests.
 *
 * @package TenUpScaffold
 */

/**
 * Dependencies
 */
// NPM Modules.
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
// Components
import ExampleBlockEdit from '../edit';

/**
 * SnapShot test
 */
describe( '<ExampleBlockEdit />', () => {
	/**
	 * Ensure our Snapshot match's up
	 */
	describe( 'snapshot', () => {
		test( 'passed', () => {
			const snapShotTree = renderer.create(
				<ExampleBlockEdit attributes={{title: wp.i18n.__( 'Test Example Block Edit' )}} setAttributes={() => {}} />
			).toJSON();
			expect( snapShotTree ).toMatchSnapshot();
		} );
	} );

	/**
	 * Enzyme tests
	 *
	 * @link https://github.com/FormidableLabs/enzyme-matchers/tree/master/packages/jest-enzyme
	 */
	describe( 'enzyme', () => {
		const mountWrapper = mount( <ExampleBlockEdit attributes={{title: wp.i18n.__( 'Test Example Block Edit' )}} setAttributes={() => {}} /> );
		test( 'exists', () => {
			expect( mountWrapper ).toExist();
		} );
		test( 'contains div with className tenup-blocks-content-list', () => {
			expect( mountWrapper ).toContainMatchingElement( '.tenup-blocks-content-list' );
		} );
	} );
} );
