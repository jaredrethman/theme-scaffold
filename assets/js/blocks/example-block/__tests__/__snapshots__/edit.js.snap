// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExampleBlockEdit /> snapshot passed 1`] = `
<div
  className="tenup-blocks-content-list"
>
  <div
    className="components-base-control components-toggle-control"
  >
    <div
      className="components-base-control__field"
    >
      <span
        className="components-form-toggle"
      >
        <input
          checked={0}
          className="components-form-toggle__input"
          id="inspector-toggle-control-1"
          onChange={[Function]}
          type="checkbox"
        />
        <span
          className="components-form-toggle__track"
        />
        <span
          className="components-form-toggle__thumb"
        />
        <svg
          aria-hidden="true"
          className="components-form-toggle__off"
          focusable="false"
          height="6"
          role="img"
          viewBox="0 0 6 6"
          width="6"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 1.5c.8 0 1.5.7 1.5 1.5S3.8 4.5 3 4.5 1.5 3.8 1.5 3 2.2 1.5 3 1.5M3 0C1.3 0 0 1.3 0 3s1.3 3 3 3 3-1.3 3-3-1.3-3-3-3z"
          />
        </svg>
      </span>
      <label
        className="components-toggle-control__label"
        htmlFor="inspector-toggle-control-1"
      >
        Toggle to test useState()
      </label>
    </div>
  </div>
  <div
    className="editor-rich-text block-editor-rich-text"
  >
    <div
      onFocus={[Function]}
    >
      <div
        onBlur={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <div
          className="components-autocomplete"
          onClick={[Function]}
        >
          <h3
            aria-activedescendant={null}
            aria-label=""
            aria-multiline={true}
            aria-owns={null}
            className="rich-text editor-rich-text__editable block-editor-rich-text__editable"
            contentEditable={true}
            dangerouslySetInnerHTML={
              Object {
                "__html": "Test Example Block Edit",
              }
            }
            onBlur={[Function]}
            onCompositionEnd={[Function]}
            onFocus={[Function]}
            onInput={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseUp={[Function]}
            onPaste={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
            role="textbox"
            style={
              Object {
                "whiteSpace": "pre-wrap",
              }
            }
            suppressContentEditableWarning={true}
          />
        </div>
      </div>
    </div>
  </div>
  <pre>
    assets/js/blocks/example-bock/index.js
  </pre>
  <div
    className="tenup-blocks-content-list__items"
  >
    <h3>
      Example Component:
    </h3>
    <ul>
      <li>
        <h4>
          Title One
        </h4>
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>Test Excerpt</p>",
            }
          }
        />
      </li>
      <li>
        <h4>
          Title Two
        </h4>
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "",
            }
          }
        />
      </li>
    </ul>
  </div>
</div>
`;
